src/porepy/models/momentum_balance.py

```diff
@@ -822,12 +844,79 @@ def contact_mechanics_numerical_constant(
         # The scaling factor should not be too large, otherwise the contact problem
         # may be discretized wrongly. I therefore introduce a safety factor here; its
         # value is somewhat arbitrary.
-        softening_factor = 1e-1
+        softening_factor = self.solid.contact_mechanics_scaler()
 
         val = softening_factor * shear_modulus / characteristic_distance
 
         return pp.ad.Scalar(val, name="Contact_mechanics_numerical_constant")
 
+    def contact_mechanics_open_state_characteristic(
```

Comment:
I think the placement here is fair for now; in the future, we need to come up with a more natural home for numerics-related modeling.